typings/webgme.d.ts

```diff
@@ -3143,103 +3249,7 @@ declare namespace Core {
-        getOwnAttributeNames(): GmeCommon.Name[];
-        getOwnValidAttributeNames(): GmeCommon.Name[];
-
-        getAttributeMeta(name: GmeCommon.Name): GmeCommon.AttrMeta;
-        getRegistryNames(): GmeCommon.Name[];
-        getOwnRegistryNames(): GmeCommon.Name[];
-
-        /** Set */
-        getMemberIds(setId: GmeCommon.SetId): GmeCommon.Path[];
-        getSetNames(): GmeCommon.Name[];
-        getMemberAttributeNames(setId: GmeCommon.SetId, memberId: GmeCommon.MemberId): GmeCommon.Name[];
-        getMemberAttribute(setId: GmeCommon.SetId, memberId: GmeCommon.MemberId): GmeCommon.OutAttr;
-        getEditableMemberAttribute(setId: GmeCommon.SetId, memberId: GmeCommon.MemberId, name: GmeCommon.Name): GmeCommon.OutAttr;
-        getMemberRegistryNames(setId: GmeCommon.SetId, memberId: GmeCommon.MemberId): GmeCommon.Name[];
-        getMemberRegistry(setId: GmeCommon.SetId, memberId: GmeCommon.MemberId, name: GmeCommon.Name): GmeCommon.Registry;
-        getEditableMemberRegistry(setId: GmeCommon.SetId, memberId: GmeCommon.MemberId, name: GmeCommon.Name): GmeCommon.Registry;
-
-        /** META */
-        getValidChildrenTypes(): GmeCommon.NodeId[];
-        getValildAttributeNames(): GmeCommon.Name[];
-        isValidAttributeValueOf(name: GmeCommon.Name, value: any): boolean;
-        getValidPointerNames(): GmeCommon.Name[];
-        getValidSetNames(): GmeCommon.Name[];
-        getConstraintNames(): GmeCommon.Name[];
-        getOwnConstraintNames(): GmeCommon.Name[];
-        getConstraint(name: GmeCommon.Name): Constraint;
-        toString(): string;
-
-        getCollectionPaths(name: GmeCommon.Name): GmeCommon.Path[];
-        getInstancePaths(): GmeCommon.Path[];
-        getJsonMeta(): GmeCommon.Metadata[];
-
-        isConnection(): boolean;
-        isAbstract(): boolean;
-        isLibraryRoot(): boolean;
-        isLibraryElement(): boolean;
-        getFullyQualifiedName(): GmeCommon.Name;
-        getNamespace(): GmeCommon.Name;
-
-        getLibraryGuid(): GUID;
-        getCrosscutsInfo(): GmeCommon.CrosscutsInfo;
-        getValidChildrenTypesDetailed(aspect: GmeCommon.Aspect, noFilter: boolean): GmeCommon.Dictionary<any>;
-        getValidSetMemberTypesDetailed(setName: GmeCommon.Name): { [key: string]: any };
-        getMetaTypeId(): Node | null;
-        getBaseTypeId(): Node | null;
-        isMetaNode(): boolean;
-        isTypeOf(typePath: GmeCommon.Path): boolean;
-        isValidChildOf(parentPath: GmeCommon.Path): boolean;
-        getValidChildrenIds(): GmeCommon.NodeId[];
-        isValidTargetOf(sourcePath: GmeCommon.Path, name: GmeCommon.Name): boolean;
-        getValidAspectNames(): GmeCommon.Name[];
-        getOwnValidAspectNames(): GmeCommon.Name[];
-        getAspectMeta(): GmeCommon.Metadata;
-
-        /** MixIns */
-        getMixinPaths(): GmeCommon.Path[];
-        canSetAsMixin(mixinPath: GmeCommon.Path): boolean;
-        isReadOnly(): boolean;
-
-    }
+    export type Node = object;
```

Comment:
Hmm... This isn't great since it will result in the following code no longer throwing a type error:
```
const notANode = {};
const name = core.getAttribute(notANode, 'name')
```

Reply:
This is a good point and I agree. Do you know if there is a way to define a type w/o specifying anything about it, but at the same time constrain anyone from passing anything that isn't of that type.
These `Core.Node`s are only being returned from a core instance.